Library/Homebrew/utils/tty.rb

```diff
@@ -56,7 +56,7 @@ def reset_escape_sequence!
   end
 
   def to_s
-    return "" unless $stdout.tty?
+    return "" unless $stdout.tty? && !ENV["NO_COLOR"]
```

Comment:
Change this into an `if` (two, if necessary) and check the `HOMEBREW_NO_COLOR` variable instead (this will still accept `NO_COLOR` but be converted elsewhere).